ChildProject/pipelines/metrics.py

```diff
@@ -472,6 +469,156 @@ def add_parser(subparsers, subcommand):
         )
 
 
+class PeriodMetrics(Metrics):
+    SUBCOMMAND = "period"
+
+    def __init__(
+        self,
+        project: ChildProject.projects.ChildProject,
+        set: str,
+        period: str,
+        period_origin: str = None,
+        get_segments: str = "vtc",
+        recordings: Union[str, List[str], pd.DataFrame] = None,
+        from_time: str = None,
+        to_time: str = None,
+        by: str = "recording_filename",
+        threads: int = 1,
+    ):
+
+        super().__init__(project, by, recordings, from_time, to_time)
+
+        self.set = set
+        self.threads = int(threads)
+
+        self.period = period
+        self.period_origin = period_origin
+
+        if self.set not in self.am.annotations["set"].values:
+            raise ValueError(
+                f"'{self.set}' was not found in the index; "
+                "check spelling and make sure the set was properly imported."
+            )
+
+        self.periods = pd.date_range(
+            start=datetime.datetime(1900, 1, 1, 0, 0, 0, 0),
+            end=datetime.datetime(1900, 1, 2, 0, 0, 0, 0),
+            freq=self.period,
+            closed="left",
+        )
+
+    def _process_unit(self, unit: str):
+        annotations, segments = self.retrieve_segments([self.set], unit)
+
+        segments = self.am.get_segments_timestamps(segments, ignore_date=True)
+        segments.dropna(subset=["onset_time"], inplace=True)
+
+        segments["onset_time"] -= pd.to_timedelta(
```

Comment:
hmm i'm lost here, it would be great to have a comment explaining what this section does...

Reply:
> > Another thing that we may want to take care of: LENA recordings can have pauses; in this case, calculating `onset_time` from `start_time` and `segment_onset` will lead to incorrect values; it is necessary, in such cases, to rely on LENA values rather than values calculated from the annotations+metadata. I suggest that the LENA converter should import the timestamps too, and that users should be able to use imported timestamps instead. What do you think @alecristia?
> 
> Yes, I thought when there are pauses in LENA recs, we split up the long .wav, and then impute the onset times of each non-pause using information from the LENA .its. Is that not how things are working right now?

We do that for EL1000, but it is possible that some people decide to do otherwise.

Reply:
> hmm i'm lost here, it would be great to have a comment explaining what this section does...
Sure, sorry. I just added more comments, does it help?

Reply:
RE allowing others to choose not to split the recording, let's not! I think it'll make data more heterogeneous in a non-interesting way.
I believe the reason why you had suggested not enforcing it had to do with recordings where there are uncertainties as to onset time, as in Vanuatu -- but that never happens with LENA recordings.
So I think it's safe to enforce it for LENA.
Would you like us to think about it some more in the case of non-LENA recs with timing uncertainty?

Reply:
This was only one of the arguments for not enforcing, the other being that this makes it harder to convert LENA datasets and to maintain them if all annotations are based on the whole, "unsplit" recordings.
But it sounds reasonable to decide, at least for some time or until someone else offers to do it, to restrict this pipeline to "contiguous" recordings.

To answer your last question: for now, I think the package should assume that the recordings are contiguous. This means that Vanuatu's case could be treated either of two ways:

 1. Using the package pipeline upstream, and then accounting for the uncertainty in the time bins in the model.
 2. Or maybe developing a different pipeline specific to this case.